src/main/java/org/kohsuke/github/HttpException.java

```diff
@@ -0,0 +1,119 @@
+package org.kohsuke.github;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Http exception
+ *
+ * @author <a href="mailto:cleclerc@cloudbees.com">Cyrille Le Clerc</a>
+ */
+public class HttpException extends IOException {
+    static final long serialVersionUID = 1L;
+
+    private final int responseCode;
+    private final String responseMessage;
+    private final String url;
+
+    /**
+     * @param message         The detail message (which is saved for later retrieval
+     *                        by the {@link #getMessage()} method)
+     * @param responseCode    Http response code. {@code -1} if no code can be discerned.
+     * @param responseMessage Http response message
+     * @param url             The url that was invoked
+     * @see HttpURLConnection#getResponseCode()
+     * @see HttpURLConnection#getResponseMessage()
+     */
+    public HttpException(String message, int responseCode, String responseMessage, String url) {
+        super(message);
+        this.responseCode = responseCode;
+        this.responseMessage = responseMessage;
+        this.url = url;
+    }
+
+    /**
+     * @param message         The detail message (which is saved for later retrieval
+     *                        by the {@link #getMessage()} method)
+     * @param responseCode    Http response code. {@code -1} if no code can be discerned.
+     * @param responseMessage Http response message
+     * @param url             The url that was invoked
+     * @param cause           The cause (which is saved for later retrieval by the
+     *                        {@link #getCause()} method).  (A null value is permitted,
+     *                        and indicates that the cause is nonexistent or unknown.)
+     * @see HttpURLConnection#getResponseCode()
+     * @see HttpURLConnection#getResponseMessage()
+     */
+    public HttpException(String message, int responseCode, String responseMessage, String url, Throwable cause) {
+        super(message, cause);
+        this.responseCode = responseCode;
+        this.responseMessage = responseMessage;
+        this.url = url;
+    }
+
+    /**
+     * @param message         The detail message (which is saved for later retrieval
+     *                        by the {@link #getMessage()} method)
+     * @param responseCode    Http response code. {@code -1} if no code can be discerned.
+     * @param responseMessage Http response message
+     * @param url             The url that was invoked
```

Comment:
I discussed this subject with the owner of the code base, and the owner has agreed that they personally will start making an effort to use the `edu` one.

I will merge the PR on my ASL clean room version of find bugs annotations tomorrow and release into central (without `@Deprecacted`)

What you choose to do is up to you

Sent from my iPhone

> On 7 Mar 2016, at 20:26, Cyrille Le Clerc <notifications@github.com> wrote:
> 
> In src/main/java/org/kohsuke/github/HttpException.java:
> 
> > +     * @see HttpURLConnection#getResponseCode()
> > +     * @see HttpURLConnection#getResponseMessage()
> > +     */
> > +    public HttpException(String message, int responseCode, String responseMessage, String url, Throwable cause) {
> > +        super(message, cause);
> > +        this.responseCode = responseCode;
> > +        this.responseMessage = responseMessage;
> > +        this.url = url;
> > +    }
> > +
> > +    /**
> > +     * @param message         The detail message (which is saved for later retrieval
> > +     *                        by the {@link #getMessage()} method)
> > +     * @param responseCode    Http response code. {@code -1} if no code can be discerned.
> > +     * @param responseMessage Http response message
> > +     * @param url             The url that was invoked
> @stephenc thanks, I am a bit stuck, net.sourceforge.findbugs:annotations is LGPL and thus incompatible with the MIT license of this project. com.github.stephenc.findbugs:findbugs-annotations is ASL2 and thus could be OK.
> 
> I'll wait for the owner of the code base to make a suggestion :-)
> 
> —
> Reply to this email directly or view it on GitHub.
>